etl/steps/data/garden/climate/2025-02-12/sst.py

```diff
@@ -0,0 +1,52 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("sst")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow.read("sst")
+
+    #
+    # Process data.
+    #
+
+    tb["nino_classification"] = tb.apply(classify_nino_anomaly, axis=1)
+    for col in ["nino_classification"]:
+        tb[col].metadata.origins = tb["nino3_4_anomaly"].metadata.origins
```

Comment:
This is a bit slow. An alternative that would propagate metadata automatically would be something like:

```
    # Fill NA values with a number that won't affect classification
    tb["oni_anomaly"] = tb["oni_anomaly"].fillna(0)

    # Classify NINO3.4 and NINO4 anomaly values
    tb["nino_classification"] = tb["oni_anomaly"].copy()
    tb.loc[tb["oni_anomaly"] > 0.5, "nino_classification"] = 1
    tb.loc[tb["oni_anomaly"] < -0.5, "nino_classification"] = 2
    tb.loc[(tb["oni_anomaly"] >= -0.5) & (tb["oni_anomaly"] <= 0.5), "nino_classification"] = 0
    tb["nino_classification"] = tb["nino_classification"].astype(int)
```